rules-reviewed/eap7/eap6/hsearch.windup.xml

```diff
@@ -0,0 +1,880 @@
+        </rule>
+        <rule id="hsearch-00109">
+            <when>
+                <javaclass references="org.hibernate.search.annotations.Key">
+                    <location>ANNOTATION</location>
+                </javaclass>
+            </when>
+            <perform>
+                <hint title="Deprecated annotation org.hibernate.search.annotations.Key in Hibernate Search 5" effort="1" severity="optional">
+                    <message>Custom filter cache keys are a deprecated feature scheduled to be removed in Hibernate Search 6. 
+                    As of Hibernate Search 5.1, filter cache keys will be determinated automatically based on the filter parameters. 
+                    Custom filter cache key methods should therefore be removed.</message>
+                    <link
+                        href="https://access.qa.redhat.com/documentation/en/red-hat-jboss-enterprise-application-platform/version-7.0/migration-guide/#migrate_hibernate_search_changes"
+                        title="Changes in Hibernate Search 5.x" />
+                    <tag>hibernate-search</tag>
+                </hint>
+            </perform>
+        </rule>
+        <rule id="hsearch-00110">
+            <when>
+                <javaclass references="org.hibernate.search.backend.configuration.impl.IndexWriterSetting.{constant}" />
+            </when>
+            <perform>
+                <hint title="Deprecated Enum Constants in Hibernate Search 5" effort="1" severity="optional">
+                    <message>The `IndexWriterSetting.MAX_THREAD_STATES` and `IndexWriterSetting.TERM_INDEX_INTERVAL` enum constants are deprecated. 
+                    They affect which properties are read from the configuration, so the fact they they are missing means that configuration properties such as `hibernate.search.Animals.2.indexwriter.term_index_interval = default` are now ignored. 
+                    The only side effect is that the property is not applied.</message>
+                    <link
+                        href="https://access.qa.redhat.com/documentation/en/red-hat-jboss-enterprise-application-platform/version-7.0/migration-guide/#migrate_miscellaneous_hibernate_search_changes"
+                        title="Miscelanous Changes in Hibernate Search 5.x" />
+                    <tag>hibernate-search</tag>
+                </hint>
+            </perform>
+            <where param="constant">
+                <matches pattern="(MAX_THREAD_STATES|TERM_INDEX_INTERVAL)" />
+            </where>
+        </rule>
+        <rule id="hsearch-00111">
+            <when>
+                <javaclass references="org.hibernate.search.annotations.Spatial" >
+                    <location>ANNOTATION</location>
+                </javaclass>
+                <!-- This doesn't find annotation with enum element specified -->
+<!--                 <javaclass references="org.hibernate.search.annotations.SpatialMode.GRID" />                 -->
```

Comment:
There should be possible to catch this:
`@Spatial(spatialMode = SpatialMode.GRID)`